contributor-docs/testing.md

```diff
@@ -5,19 +5,88 @@
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 ## Table of Contents
 
-- [Overview](#overview)
-- [Prerequisites](#prerequisites)
-- [Components](#components)
-  - [Visual Regression Testing](#visual-regression-testing)
-  - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Strategy](#strategy)
+- [Unit Tests](#unit-tests)
+  - [Overview](#overview)
+  - [What We Test](#what-we-test)
+  - [Running Tests](#running-tests)
+- [Visual Regression Tests](#visual-regression-tests)
+  - [Overview](#overview)
+  - [Prerequisites](#prerequisites)
+  - [Components](#components)
+    - [Visual Regression Testing](#visual-regression-testing)
+    - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Interaction Tests](#interaction-tests)
+  - [As A Part Of Unit Tests](#as-a-part-of-unit-tests)
+  - [Storybook Interaction Tests](#storybook-interaction-tests)
+- [Accessibility Tests](#accessibility-tests)
 - [Continous Integration](#continous-integration)
 - [FAQ](#faq)
   - [Why am I seeing `browserType.launch: Executable doesn't exist at ../path`?](#why-am-i-seeing-browsertypelaunch-executable-doesnt-exist-at-path)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 <!-- prettier-ignore-end -->
 
-## Overview
+##Strategy
+
+Strategy
+
+##Unit Tests
+
+###Overview
+
+We use [Jest](https://jestjs.io/) as our test runner to write and run our unit tests. Our prefered way of structuring tests is with `describe/it` block format and they are authored within the components's source directory with the file name of `componentName.test.tsx`
+
+We predominantly use [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/)'s testing helpers such as
+
+- [render](https://testing-library.com/docs/react-testing-library/api#render) to render components
+- [userEvent](https://testing-library.com/docs/user-event/intro/) or [Event API](https://testing-library.com/docs/dom-testing-library/api-events/) to fire events and simulate user interaction on the component\*
+- `screen`
+- `act`
+
+To make assertions about the elements we use [Jest](https://jestjs.io/) and [jest-dom](https://github.com/testing-library/jest-dom).
+
+\*: You can read about the differences between `fireEvent` and `UserEvent` [here](https://testing-library.com/docs/user-event/intro/#differences-from-fireevent).
+
+###What We Test
+
+We try to achieve high coverage on unit tests and make sure to test components' API such as their props, events and callback functions.
```

Comment:
I am glad you mentioned that Josh, because I think this needs to be unpacked and 100% agree about setting a baseline here if we are going to use them as a metric.

I spoke this with @colebemis and I think with @siddharthkp as well if I don't remember wrong. From what I gathered, they might be useful as a reference point but because we do not have much clarity around what is expected, it feels a bit ambiguous to mention about unit test coverage in the docs. 

Would be great to hear what others think 👀